File: ((((Practice Problems))))/Weak Student/CPP/3.PersonMain/PersonMain.cpp
#include<iostream>
using namespace std;


class Person
{
public:
   int id;
   char name[20],city[20];

    void setPerson()
    {
        cout<<"Enter Id : ";
        cin>>id;
        cout<<"Enter Name : ";
        cin>>name;
        fflush(stdin);
        cout<<"Enter City : ";
        cin>>city;
    }

    void getPerson()
    {
        cout<<"ID : "<<id<<endl;
        cout<<"Name : "<<name<<endl;
        cout<<"City : "<<city<<endl;
    }
};

class Student : public Person
{
public:
    char branch[20];
    int sem,m1,m2;

    void setStudent()
    {
        setPerson();
        cout<<"Enter Branch : ";
        cin>>branch;
        fflush(stdin);
        cout<<"Enter Semester : ";
        cin>>sem;
        cout<<"Enter Marks of Subject 1 : ";
        cin>>m1;
        cout<<"Enter Marks of Subject 2 : ";
        cin>>m2;
    }

    void getStudent()
    {
        getPerson();
        cout<<"Branch : "<<branch<<endl;
        cout<<"Semester : "<<sem<<endl;
        cout<<"Marks of Subject 1 : "<<m1<<endl;
        cout<<"Marks of Subject 2 : "<<m2<<endl;
    }

    void avgMarks()
    {
        cout<<"Average Marks are : "<<(((float)m1+m2)/2);
    }
};

class VisitingTeacher : public Person
{
public:
    int exp,rph,nhrs;

    void setVisitingTeacher()
    {
        setPerson();
        cout<<"Enter Experience : ";
        cin>>exp;
        cout<<"Enter Rate Per Hour : ";
        cin>>rph;
        cout<<"Enter Number of Hours : ";
        cin>>nhrs;
    }


    void getVisitingTeacher()
    {
        getPerson();
        cout<<"Experience : "<<exp<<endl;
        cout<<"Rate per Hour : "<<rph<<endl;
        cout<<"Number of Hours : "<<nhrs<<endl;
    }

    void calSalary()
    {
        cout<<"Monthly Salary : "<<(rph*nhrs);
    }
};

int main()
{
    VisitingTeacher t;
    t.setVisitingTeacher();
    t.getVisitingTeacher();
    t.calSalary();

    Student s;
    s.setStudent();
    s.getStudent();
    s.avgMarks();


}

File: 57.PT1 Q3/PersonMain.java
import java.util.Scanner;
class Person
{
	int id;
	String name,city;
	Person(int id,String name,String city)
	{
		this.id=id;
		this.name=name;
		this.city=city;
	}
	void getPerson()
	{
		System.out.println("Id : "+id);
		System.out.println("Name : "+name);
		System.out.println("City : "+city);
	}
}
class VisitingTeacher extends Person
{
	int exp,rph;
	VisitingTeacher(int id,String name,String city,int exp,int rph)
	{
		super(id,name,city);
		this.exp=exp;
		this.rph=rph;
	}
	void getTeacher()
	{
		getPerson();
		System.out.println("Exp : "+exp);
		System.out.println("Rate Per Hour : "+rph);
	}
	float calSalary()
	{
		Scanner sc=new Scanner(System.in);
		int nhrs;
		System.out.print("Enter Number of Hours : ");
		nhrs=sc.nextInt();
		if(nhrs<40)
		{
			return nhrs*rph;
		}
		else
		{
			return (40*rph)+((nhrs-40)*rph*1.5f);
		}
	}
}
class Pupil extends Person
{
	int sem;
	String branch;
	int sub[],n;
	Pupil(int id,String name,String city,String branch,int sem)
	{
		super(id,name,city);
		this.branch=branch;
		this.sem=sem;
		Scanner sc =new Scanner(System.in);
		System.out.print("Enter Number of Subjects : ");
		n=sc.nextInt();
		sub=new int [n];
		for(int i=0;i<n;i++)
		{
			System.out.print("Enter Marks of Subject "+(i+1)+" : ");
			sub[i]=sc.nextInt();
		}
	}
	float avgMarks()
	{
		int sum=0;
		for(int i=0;i<sub.length;i++)
		{
			sum=sum+sub[i];
		}
		return (float)sum/sub.length;
	}
	void getPupil()
	{
		getPerson();
		System.out.println("Semester : "+sem);
		System.out.println("Branch : "+branch);
		for(int i=0;i<n;i++)
		{
			System.out.println("Marks of Subject "+(i+1)+" : "+sub[i]);
		}
	}
}
class PersonMain
{
	public static void main(String args[])
	{
		//Person p=new Person(1,"Mihir","Mumbai");
		//p.getPerson();
		System.out.println("\nVisiting Teacher : \n");
		VisitingTeacher v=new VisitingTeacher(1,"Mihir","Mumbai",1,500);
		v.getTeacher();
		//v.calSalary();
		System.out.println("Salary of Teacher : "+v.calSalary());
		System.out.println("\nPupil : \n");
		Pupil p=new Pupil(1,"Mihir","Mumbai","IT",3);
		p.getPupil();
		System.out.println("Average Marks : "+p.avgMarks());
	}
}